cabot/metricsapp/api/elastic_grafana.py

```diff
@@ -0,0 +1,179 @@
+import logging
+import json
+from elasticsearch_dsl import Search, A
+from elasticsearch_dsl.query import Range
+from django.core.exceptions import ValidationError
+from cabot.metricsapp.defs import ES_SUPPORTED_METRICS
+
+
+logger = logging.getLogger(__name__)
+
+
+def build_query(series, min_time='now-10m', default_interval='1m'):
+    """
+    Given series information from the Grafana API, build an Elasticsearch query
+    :param series: a "target" in the Grafana dashboard API response
+    :param min_time: the earliest time we're looking for
+    :return: Elasticsearch query
+    """
+    search = Search().query('query_string', query=series['query'], analyze_wildcard=True) \
+        .query(Range(** {series['timeField']: {'gte': min_time}}))
+    search.aggs.bucket('agg', get_aggs(series, min_time, default_interval))
+    return search.to_dict()
+
+
+def get_aggs(series, min_time, default_interval):
+    """
+    Get the ES aggregations from the input Grafana API series info
+    :param series: a "target" in the Grafana dashboard API response
+    :return an elasticsearch-dsl "agg" containing all the aggregation info
+    """
+    aggs = None
+    # Variable to chain all the aggregations to. When a new agg is added with .bucket, that
+    # agg is returned, so "aggs" will keep track of the first aggregation and "aggs_chain" will
+    # keep track of the most recently added one.
+    aggs_chain = None
+    date_histogram = None
+
+    for agg in series['bucketAggs']:
+        # date_histogram must be the final aggregation--save it to add after the other aggregations
+        type = agg['type']
+        if type == 'date_histogram':
+            date_histogram = agg
+            continue
+
+        # Add extra settings for "terms" aggregation
+        elif type == 'terms':
+            settings = get_terms_settings(agg)
+
+            if aggs is None:
```

Comment:
That doesn't work (A has to have arguments), but I made a different change to get rid of the if statement. Let me know if it looks ok!